File: 프로그래머스/1/42889. 실패율/실패율.py
def solution(N, stages):
    answer = []
    fail = []
    counter = 0
    
    for n in range(1, N + 1):
        cnt = 0
        for stage in stages:
            if stage == n:
                cnt += 1
        
        counter += cnt
        if n == 1:
            fail.append({'stage': n, 'fail_p': cnt / len(stages), 'cnt_s': counter})
        else:
            remaining_players = len(stages) - fail[n - 2]['cnt_s']
            fail_rate = cnt / remaining_players if remaining_players > 0 else 0
            fail.append({'stage': n, 'fail_p': fail_rate, 'cnt_s': counter})

    fail.sort(key=lambda x: x['fail_p'], reverse=True)
    
    for n in range(N):
        answer.append(fail[n]['stage'])
    
    return answer

File: 프로그래머스/1/42889. 실패율/실패율.js
function solution(N, stages) {
    var answer = [];
    let fail = [];
    let cnter = 0;
    for(let i =1; i<N+1;i++){
        let cnt =0;
        for(let j=0;j<stages.length;j++){
            if(stages[j] === i){
                cnt=cnt+1      
            }
        }
        
        cnter =cnter + cnt;
        if (i === 1){
            fail.push({fir: i,sec:cnt/stages.length,thr: cnter})
            }

        if(i !== 1) {
            fail.push({fir: i,sec: cnt/(stages.length-fail[i-2].thr),thr: cnter});
        }
    }

    fail.sort(function(a, b)  {
        return b.sec - a.sec;
    });
    for(let q=0;q<N;q++){
        answer.push(fail[q].fir)
    }
    return answer;
}